### PR TITLE
Poll the chain head to detect new blocks instead of using a filter

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
eth_newBlockFilter is not supported by Infura over HTTP and is actually less efficient as it winds up polling more frequently and doesn't skip blocks if the chain progresses by more than one block in the polling period.

Instead of the filter, we now just request the latest block periodically.

### CODE REVIEW

File: pow/src/main/java/tech/pegasys/teku/pow/Eth1HeadTracker.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            error -> {
+              LOG.warn("Failed to get latest ETH1 chain head. Will retry.", error);
+              return null;
+            })
+        .always(
+            () ->
+                asyncRunner
+                    .runAfterDelay(
+                        this::pollLatestHead,
+                        Constants.SECONDS_PER_ETH1_BLOCK.longValue(),
```
Comment (cemozerr) [inline pow/src/main/java/tech/pegasys/teku/pow/Eth1HeadTracker.java:67]: There is a possibility that running `pollLatestHead` every `SECONDS_PER_ETH1_BLOCK` leads to us trailing almost a block behind the `ETH1_FOLLOW_DISTANCE`. Not sure if that would be a problem.

File: pow/src/main/java/tech/pegasys/teku/pow/Eth1Provider.java [hunk 2]
```diff
@@ -31,7 +28,7 @@
 
   SafeFuture<Block> getGuaranteedEth1BlockFuture(UnsignedLong blockNumber);
 
-  SafeFuture<Block> getLatestEth1BlockFuture();
+  SafeFuture<Block> getLatestEth1Block();
```
Comment (cemozerr) [inline pow/src/main/java/tech/pegasys/teku/pow/Eth1Provider.java:31]: nit: to keep the method names in line, you might want to either add Future to the end of this method name or remove all the Futures from the method names above.

**Review by cemozerr — APPROVED** [review]
LGTM!

File: pow/src/main/java/tech/pegasys/teku/pow/Eth1Provider.java (hunk 2 above)
Comment (ajsutton) [reply pow/src/main/java/tech/pegasys/teku/pow/Eth1Provider.java:31]: I had meant to remove future from them all but forgot to do a full pass will fix.

File: pow/src/main/java/tech/pegasys/teku/pow/Eth1HeadTracker.java (hunk 1 above)
Comment (ajsutton) [reply pow/src/main/java/tech/pegasys/teku/pow/Eth1HeadTracker.java:67]: I think we can live with that - we're already 1024 blocks behind the chain, what's one more?  And in practice block Tims are never exactly `SECONDS_PER_ETH1_BLOCK` apart anyway (on Görli block are consistently every 15 seconds, not 14 and the vary a lot on Mainnet) so the amount we're behind will vary.

**Pull request merged**